index.js

```diff
@@ -108,8 +108,10 @@ function unwrapReturns(src, result) {
       }
     },
     ReturnStatement: function (node) {
-      hasReturn = true
-      replace(node, 'return {value: ' + source(node.argument) + '};');
+      hasReturn = true;
+      if (node.argument) {
+        replace(node, 'return {value: (' + source(node.argument) + ')};');
```

Comment:
Even if there's no argument, we still need to re-write to return `{value: undefined}`.  Otherwise the code generated by `with` won't be able to tell that a return has happened.